xformers/components/attention/csrc/cuda/mem_eff_attention/kernel_backward.h

```diff
@@ -110,6 +110,23 @@ struct AttentionBackwardKernel {
       grad_query_ptr += batch_id * head_dim * num_queries;
       grad_key_ptr += batch_id * head_dim * num_keys;
       grad_value_ptr += batch_id * head_dim_value * num_keys;
+
+      head_dim = warp_uniform(head_dim);
+      head_dim_value = warp_uniform(head_dim_value);
+      num_queries = warp_uniform(num_queries);
+      num_keys = warp_uniform(num_keys);
+
+      query_ptr = warp_uniform(query_ptr);
+      key_ptr = warp_uniform(key_ptr);
+      value_ptr = warp_uniform(value_ptr);
+      logsumexp_ptr = warp_uniform(logsumexp_ptr);
+      output_ptr = warp_uniform(output_ptr);
+      grad_output_ptr = warp_uniform(grad_output_ptr);
+      delta_ptr = warp_uniform(delta_ptr);
+
+      grad_query_ptr = warp_uniform(grad_query_ptr);
+      grad_key_ptr = warp_uniform(grad_key_ptr);
+      grad_value_ptr = warp_uniform(grad_value_ptr);
```

Comment:
note to self: this trick is needed otherwise there would be a performance slowdown